vowpalwabbit/red_python.h

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
+// individual contributors. All rights reserved. Released under a BSD (revised)
+// license as described in the file LICENSE.
+
+#pragma once
+#include <vector>
+#include "reductions_fwd.h"
+
+VW::LEARNER::base_learner* red_python_setup(VW::config::options_i& options, vw& all);
+
+namespace RED_PYTHON
+{
+    struct PythonCppBridge
```

Comment:
Is it possible for this to be only exposed in the python lib? Perhaps the reduction could be a generic hook reduction that allows the consumer (python bindings) to define custom logic?

Reply:
good point

Reply:
I refactored this part to make it more flexible, in theory it's been moved to the python side so this could be replicated with any other binding.